File: 1127-last-stone-weight/1127-last-stone-weight.py
class Solution(object):
    def k(self,l):
        if len(l)==1:
            return l[0]
        l[len(l)-2]=abs(l[len(l)-1]-l[len(l)-2])
        l=l[:len(l)-1]
        return self.k(sorted(l))
    def lastStoneWeight(self, stones):
        l=stones
        l=sorted(l)
        return self.k(l)
        

File: 1127-last-stone-weight/1127-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int num:stones){
            pq.add(num);
        }
        while(pq.size()>1){
            int p = pq.poll();
            int q = pq.poll();
            if(p!=q){
                pq.add(p-q);
            }
        }
        if(pq.isEmpty()) return 0;
        return pq.poll();
    }
}